src/CoreDumpAnalysis/analysis/DebugSymbolResolver.cs

```diff
@@ -16,83 +18,77 @@ public class DebugSymbolResolver {
+				.Select(g => g.First());
 		}
 
-		private void DownloadDebugSymbols(IList<SDModule> modules) {
-			foreach(SDModule module in modules) {
+		private async Task DownloadDebugSymbolsAsync(IEnumerable<SDModule> modules) {
+			foreach (SDModule module in modules) {
 				if (module is SDCDModule) {
-					DownloadDebugSymbolForModule((SDCDModule)module);
+					await DownloadDebugSymbolForModuleAsync((SDCDModule)module);
 				} else {
 					throw new InvalidCastException("Provided module is not a coredump module!");
 				}
 			}
 		}
 
-		private void DownloadDebugSymbolForModule(SDCDModule module) {
-			if (module.LocalPath != null && DebugSymbolsRelevant(module)) {
+		private async Task DownloadDebugSymbolForModuleAsync(SDCDModule module) {
+			if (module.LocalPath != null && IsDynatraceModule(module)) {
 				string hash = filesystem.Md5FromFile(module.LocalPath);
 				if (IsDebugFileAvailable(module, hash)) {
 					module.DebugSymbolPath = Path.GetFullPath(DebugFilePath(module.LocalPath, hash));
 				} else {
-					DownloadDebugSymbols(module, hash);
+					await DownloadDebugSymbolsAsync(module, hash);
 				}
 			}
 		}
 
-		private bool DebugSymbolsRelevant(SDCDModule module) {
-			if(module.FileName.Contains("libruxit") || module.FileName.Contains("liboneagent")) {
+		private bool IsDynatraceModule(SDCDModule module) {
+			if (module.FileName.Contains("libruxit") || module.FileName.Contains("liboneagent")) {
 				return true;
 			}
-			if(module.FilePath.Contains("/lib/ruxit") || module.FilePath.Contains("/lib64/ruxit") || 
+			if (module.FilePath.Contains("/lib/ruxit") || module.FilePath.Contains("/lib64/ruxit") ||
 				module.FilePath.Contains("/lib/oneagent") || module.FilePath.Contains("/lib64/oneagent")) {
 				return true;
 			}
 			return false;
 		}
-		
+
 		private bool IsDebugFileAvailable(SDCDModule module, string hash) {
 			return filesystem.FileExists(DebugFilePath(module.LocalPath, hash));
 		}
 
-		private void DownloadDebugSymbols(SDCDModule lib, string hash) {
-			Console.WriteLine("Trying to retrieve debug symbols for " + lib.FilePath);
+		private async Task DownloadDebugSymbolsAsync(SDCDModule lib, string hash) {
+			Console.WriteLine($"Trying to retrieve debug symbols for {lib.FilePath}");
 			string url = Constants.DEBUG_SYMBOL_URL_PATTERN.Replace("{hash}", hash).Replace("{file}", DebugFileName(lib.LocalPath));
 
 			string localDebugFile = DebugFilePath(lib.LocalPath, hash);
 			try {
-				if (requestHandler.DownloadFromUrl(url, localDebugFile)) {
-					Console.WriteLine("Successfully downloaded debug symbols for " + lib.FilePath + ". Stored at " + localDebugFile);
-					lib.DebugSymbolPath = Path.GetFullPath(localDebugFile);
-				} else {
-					Console.WriteLine("Failed to download debug symbols for " + lib.FilePath + ". URL: " + url);
-				}
-			} catch(Exception e) {
-				Console.WriteLine("Failed to download debug symbol: " + e.Message);
+				await requestHandler.DownloadFromUrlAsync(url, localDebugFile)
```

Comment:
if (await requestHandler.DownloadFromUrlAsync(url, localDebugFile)) {
					Console.WriteLine($"Successfully downloaded debug symbols for {lib.FilePath}. Stored at {localDebugFile}");
					lib.DebugSymbolPath = Path.GetFullPath(localDebugFile);
				} else {
					Console.WriteLine($"Failed to download debug symbols for {lib.FilePath}. URL: {url}");
				}